hack/verify-receipts-upgrade-migration.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script tests "krew system receipts-migration" which was introduced for
+# migrating krew 0.2.x to krew 0.3.x.
+#
+# TODO(ahmmetb,corneliusweig) remove at/after krew 0.4.x when receipts-migration
+# is no longer supported.
+
+set -euo pipefail
+
+[[ -n "${DEBUG:-}" ]] && set -x
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BINDIR="${SCRIPTDIR}/../out/bin"
+goos="$(go env GOOS)"
+goarch="$(go env GOARCH)"
+
+install_krew_0_2_1() {
+    # from https://github.com/kubernetes-sigs/krew/blob/v0.2.1/README.md
+    krew_root="${1}"
+    cd "$(mktemp -d)"
+    curl -fsSLO "https://storage.googleapis.com/krew/v0.2.1/krew.{tar.gz,yaml}"
+    tar zxf krew.tar.gz
+    env KREW_ROOT="${krew_root}" ./krew-"${goos}_amd64" install \
+        --manifest="./krew.yaml" --archive="./krew.tar.gz"
+}
+
+install_plugin() {
+    krew_root="${1}"
+    plugin="${2}"
+
+    run_krew "${1}" install "${plugin}" 1>/dev/null
+}
+
+# patch_krew_bin replaces the installed krew 0.2.x binary with the specified
+patch_krew_bin() {
+    krew_root="${1}"
+    new_binary="${2}"
+
+    local old_binary
+    old_binary="$(readlink -f "${krew_root}/bin/kubectl-krew")"
+    cp -f "${new_binary}" "${old_binary}"
+}
+
+# run_krew runs 'krew' with the specified KREW_ROOT and arguments.
+run_krew() {
+    krew_root="${1}"
+    shift
+
+    env KREW_ROOT="${krew_root}" \
+        PATH="${krew_root}/bin:$PATH" \
+        kubectl krew "$@"
+}
+
+verify_plugin_installed() {
+    krew_root="${1}"
+    plugin="${2}"
+    local plugin_bin_name
+    plugin_bin_name="kubectl-${plugin/-/_}"
```

Comment:
```suggestion
    plugin_bin_name="kubectl-${plugin//-/_}"
```
Otherwise only the first occurrence of `-` will be replaced